### ISSUE TITLE
EncryptedFile examples containing mimetype

### ISSUE THREAD

**Issue by deepbluev7** [matrix-org/matrix-doc#2374]
While looking at the spec of the [EncryptedFile](https://matrix.org/docs/spec/client_server/latest#extensions-to-m-message-msgtypes) needed for attachment encryption, I noticed that all the examples contain a mimetype field, which is duplicated from the info subobject. The specification for EncryptedFile doesn't list that as an optional member though. Riot also seems to send this additional mimetype. So I'm wondering, should that be in the spec, are the examples wrong and why even is that field duplicated?

**Comment by uhoreg** [comment]
The original intention was that the mimetype is supposed to be duplicated, but was accidentally omitted when that was originally merged.  However, [MSC2582](https://github.com/matrix-org/matrix-doc/pull/2582) is a proposal to drop the field.

**Comment by joepie91** [comment]
Following on from #2699:

> (Since this is in the encrypted part of the event, this has nothing to do with Synapse.)

I'm not sure I'm following. The property in question is not, itself, encrypted - it just describes the data that *is*. Doesn't that make it something that the homeserver should be concerned about the validity of?

**Comment by deepbluev7** [comment]
No, this applies only to encrypted events and since this is in content and not a relation, it is encrypted and the homeserver can't see it at all. `file` is only ever used in encrypted events, since it specifies how to decrypt it, so the homeserver can't ever validate it. The homeserver can only check, if it is sent unencrypted by accident.

**Comment by joepie91** [comment]
For posterity: after a brief chat with @deepbluev7 it turned out that the bit I overlooked was that the *entire* media event is getting encrypted, and so the homeserver never sees *any* of it, and that is why it can't be validated. It's clear now :)
